dcp-0012/dcp-0012.mediawiki

```diff
@@ -13,5 +13,436 @@ Replaces: DCP0010
+// provided height.  This value is reduced over time based on the height and
+// then split proportionally between PoW, PoS, and the Treasury.
+func calcBlockSubsidy(height int64) int64 {
+	// These parameters are ordinarily unique per chain and thus should be
+	// passed into the function via a chain parameters structure, however,
+	// they are defined as constants with the mainnet parameters here for the
+	// purposes of providing a self-contained function for the DCP.
+	const (
+		mainnetBlockOneSubsidy            = 1680000 * 100000000
+		mainnetSubsidyReductionInterval   = 6144
+		mainnetBaseSubsidy                = 3119582664
+		mainnetSubsidyReductionMultiplier = 100
+		mainnetSubsidyReductionDivisor    = 101
+	)
+
+	// Negative block heights are invalid and produce no subsidy.
+	// Block 0 is the genesis block and produces no subsidy.
+	// Block 1 subsidy is special as it is used for initial token distribution.
+	switch {
+	case height <= 0:
+		return 0
+	case height == 1:
+		return mainnetBlockOneSubsidy
+	}
+
+	// Calculate the subsidy by applying the appropriate number of reductions
+	// per the requested height.
+	reductions := height / mainnetSubsidyReductionInterval
+	subsidy := int64(mainnetBaseSubsidy)
+	for i := int64(0); i < reductions; i++ {
+		subsidy *= mainnetSubsidyReductionMultiplier
+		subsidy /= mainnetSubsidyReductionDivisor
+
+		// Stop once no further reduction is possible.  This ensures a bounded
+		// computation for large requested intervals and that all future
+		// requests for intervals at or after the final reduction interval
+		// return 0 without recalculating.
+		if subsidy == 0 {
+			break
+		}
+	}
+	return subsidy
+}
+</pre>
+
+===Work (PoW) Proportion Subsidy Calculation===
+
+<pre>
+// calcWorkSubsidyDCP0012 returns the proof of work subsidy for a block at the
+// provided height and given number of votes using the round 2 modified subsidy
+// split.
+func calcWorkSubsidyDCP0012(height int64, voters uint16) int64 {
+	// These parameters are ordinarily unique per chain and thus should be
+	// passed into the function via a chain parameters structure, however,
+	// they are defined as constants with the mainnet parameters here for the
+	// purposes of providing a self-contained function for the DCP.
+	const (
+		mainnetStakeValidationHeight = 4096
+		votesPerBlock                = 5
+		minVotesRequired             = (votesPerBlock / 2) + 1
+	)
+
+	// The first block has special subsidy rules.
+	if height == 1 {
+		return calcBlockSubsidy(height)
+	}
+
+	// The subsidy is zero if there are not enough voters once voting begins.  A
+	// block without enough voters will fail to validate anyway.
+	if height >= mainnetStakeValidationHeight && voters < minVotesRequired {
+		return 0
+	}
+
+	// Calculate the full block subsidy and reduce it according to the PoW
+	// proportion.
+	const proportion = 1
+	const totalProportions = 100
+	subsidy := calcBlockSubsidy(height)
+	subsidy *= proportion
```

Comment:
Right.  It's not really necessary, but it's for consistency.  It is interesting that it caused you to stop for a moment though because it personally though it was clearly to keep it than not because it makes it more obvious to me that it is 1/100 = 1%.

Reply:
Just different brains working in different ways I guess. Maybe because you are more familiar with the code. It can stay as it is, it isn't incorrect after all, I just wanted to highlight it in case it was something you missed.